src/components/slideshow/slideshow.js

```diff
@@ -276,14 +287,40 @@ export default function (options) {
         if (btnSlideshowNext) btnSlideshowNext.classList.add('hide');
     }
 
+    function showAutoplayDelayMenu(button) {
+        const delayLengths = [1, 3, 5, 10, 15];
+        const items = delayLengths.map(length => {
+            return {
+                id: length * 1000,
+                name: globalize.translate('ValueSeconds', length)
+            };
+        });
+
+        const actionSheetOptions = {
+            items,
+            positionTo: button
+        };
+
+        actionSheet.show(actionSheetOptions).then(function (result) {
+            if (result) {
+                swiperInstance.params.autoplay.delay = result;
+                swiperInstance.autoplay.start();
+                userSettings.set('autoplayDelay', result, true);
+            }
+        }).catch(() => { /* swallow errors */ });
+    }
+
     /**
      * Handles OSD changes when the autoplay is started.
      */
-    function onAutoplayStart() {
+    function onAutoplayStart(autoplayDelay) {
         const btnSlideshowPause = dialog.querySelector('.btnSlideshowPause .material-icons');
         if (btnSlideshowPause) {
             btnSlideshowPause.classList.replace('play_arrow', 'pause');
         }
+        if (autoplayDelay) {
+            swiperInstance.params.autoplay.delay = autoplayDelay;
```

Comment:
Isn't initialization enough?

Reply:
Starting with shuffle or play all then using the play/pause button works as expected without this, but if the user directly opens an image and then presses play it uses the default delay without this.

Reply:
After testing more, there's still something wrong with this. Some part of pausing the slideshow by dragging the image causes the wrong delay to be used after playing. It only happens sometimes though, so I'll keep investigating.

Reply:
Hey @MCBuckethead did you have any luck fixing the issue you mentioned here?

Reply:
Has this being abandoned? I wanted to change the default from 3 to 7 seconds per slide and it was rejected since this is a better option. But if this is going nowhere maybe setting the default to 7 sec is not a bad idea.